src/util/job-name-validation.tsx

```diff
@@ -0,0 +1,66 @@
+// Utilities to validate and sanitize job names (make them valid)
+
+import { TranslationBundle } from '@jupyterlab/translation';
+
+const jobNameRegex = /^[a-zA-Z0-9._][a-zA-Z0-9._-]{0,62}$/;
+const invalidFirstCharRegex = /^[^a-zA-Z0-9._]/;
+const invalidCharRegex = /[^a-zA-Z0-9._-]/g;
+const maxLength = 63;
```

Comment:
These are fine for now, but we should switch to a library that provides a declarative validation API rather than approaching this problem imperatively. See #234 comment.

Reply:
Filed #260 for this.